packages/data-broker/src/__tests__/services/tupaia/TupaiaService.test.ts

```diff
@@ -32,138 +32,139 @@ describe('TupaiaService', () => {
+        await tupaiaService.pullAnalytics([DATA_ELEMENTS.POP01, DATA_ELEMENTS.POP02], {
           dataServiceMapping,
           period: '20200822',
         });
-        expect(tupaiaDataApi.fetchEvents).toHaveBeenCalledOnceWith(
+        expect(tupaiaDataApi.fetchAnalytics).toHaveBeenCalledOnceWith(
           expect.objectContaining({ startDate: '2020-08-22', endDate: '2020-08-22' }),
         );
       });
 
       it('supports various API options', async () => {
         const options = {
           dataServiceMapping,
-          dataElementCodes: ['POP01', 'POP02'],
           organisationUnitCodes: ['TO', 'PG'],
           startDate: '20200731',
           endDate: '20200904',
         };
 
-        await tupaiaService.pull([DATA_GROUPS.POP01_GROUP], 'dataGroup', options);
-        expect(tupaiaDataApi.fetchEvents).toHaveBeenCalledOnceWith(
+        await tupaiaService.pullAnalytics([DATA_ELEMENTS.POP01], options);
+        expect(tupaiaDataApi.fetchAnalytics).toHaveBeenCalledOnceWith(
           expect.objectContaining(options),
         );
       });
+    });
+
+    describe('data pulling', () => {
+      it('returns a { results, metadata, numAggregationsProcessed } response', async () => {
+        const response = await tupaiaService.pullAnalytics(
+          [DATA_ELEMENTS.POP01, DATA_ELEMENTS.POP02],
+          { dataServiceMapping },
+        );
+        expect(response).toHaveProperty('results');
+        expect(response).toHaveProperty('metadata');
+        expect(response).toHaveProperty('numAggregationsProcessed');
+      });
+
+      it('returns the analytics API response in the `results` field', () =>
+        expect(
+          tupaiaService.pullAnalytics([DATA_ELEMENTS.POP01, DATA_ELEMENTS.POP02], {
+            dataServiceMapping,
+          }),
+        ).resolves.toHaveProperty('results', ANALYTICS.analytics));
 
-      it('directly returns the event API response', () =>
+      it('correctly builds the `metadata` field', () =>
+        expect(
+          tupaiaService.pullAnalytics([DATA_ELEMENTS.POP01, DATA_ELEMENTS.POP02], {
+            dataServiceMapping,
+          }),
+        ).resolves.toHaveProperty('metadata', {
+          dataElementCodeToName: {
+            POP01: 'Population 1',
+            POP02: 'Population 2',
+          },
+        }));
+
+      it('returns the analytics API aggregations processed in the `numAggregationsProcessed` field', () =>
         expect(
-          tupaiaService.pull([DATA_GROUPS.POP01_GROUP], 'dataGroup', { dataServiceMapping }),
-        ).resolves.toStrictEqual(EVENTS));
+          tupaiaService.pullAnalytics([DATA_ELEMENTS.POP01, DATA_ELEMENTS.POP02], {
+            dataServiceMapping,
+          }),
+        ).resolves.toHaveProperty('numAggregationsProcessed', ANALYTICS.numAggregationsProcessed));
+    });
+  });
+
+  describe('pullEvents', () => {
+    it('throws an error if multiple data groups are provided', () =>
```

Comment:
I duplicated some test cases for `pullAnalytics` and `pullEvents`, since they are separate APIs now